### PR TITLE
User/kusnarski/modals for adding

### PR DESCRIPTION
**Pull request by krisx303**
image
image
image
image
image

### CODE REVIEW

File: src/pages/main/NoteCard.tsx [hunk 1]
```diff
@@ -12,13 +12,12 @@ const NoteCard: React.FC<NoteCardProps> = ({ note }) => {
...
     const handlePress = () => {
-        // Navigate to the Card page
         navigation.navigate('CardPage', { noteId: note.id, workspaceId: note.workspaceId});
     };
 
     return (
-        <TouchableRipple onPress={handlePress}>
-            <Card key={note.id} style={styles.card}>
+        <TouchableRipple onPress={handlePress} style={styles.card}>
+            <Card key={note.id} style={{flex: 1}}>
```
Comment (Adam3004) [inline src/pages/main/NoteCard.tsx:20]: flex 1?

Comment (krisx303) [reply src/pages/main/NoteCard.tsx:20]: żeby rozciągnąć maxymalnie element (ale na tyle żeby wszystko inne się zmieściło)

File: src/pages/main/Types.ts [hunk 2]
```diff
@@ -8,17 +8,17 @@ export interface NoteSummary {
     workspaceId: string;
     title: string;
     summary: string;
-    author: string | undefined;
-    date: string | undefined;
+    author?: string | undefined;
```
Comment (Adam3004) [inline src/pages/main/Types.ts:11]: what it change?

File: src/pages/main/modals/CreateNoteModal.tsx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        color: 'red',
+        marginBottom: 10,
+    },
+    button: {
+        marginHorizontal: 10,
+        marginVertical: 5,
+    },
+    cancelButton: {
+        borderColor: '#7912b0',
+        color: '#7912b0',
```
Comment (Adam3004) [inline src/pages/main/modals/CreateNoteModal.tsx:174]: this class seems to be not so easy but imo is write well, good job

File: src/transport/HttpClient.ts [hunk 4]
```diff
@@ -8,6 +8,12 @@ interface HttpClientBase {
     getRecentQuizzes(): Promise<QuizSummary[]>;
 
     getQuizDetails(workspaceId: string, quizId: number): Promise<QuizDetails>;
+
+    getWorkspaces(): Promise<Workspace[]>;
+
+    verifyNoteIdIdentity(workspaceId: string, noteId: string): Promise<boolean>;
+
+    verifyQuizIdIdentity(quizId: string): Promise<boolean>;
```
Comment (Adam3004) [inline src/transport/HttpClient.ts:18]: i hope it won't be problematic to implement in the future

**Review by Adam3004 — APPROVED** [review]
looks good for me

File: src/pages/main/Types.ts (hunk 2 above)
Comment (krisx303) [reply src/pages/main/Types.ts:11]: makes author property nullable - I will revert that to be required nonnull

File: src/transport/HttpClient.ts (hunk 4 above)
Comment (krisx303) [reply src/transport/HttpClient.ts:18]: why could it be problematic? just a db query for existance by id. should be pretty easy

**Pull request merged**